SCClassLibrary/Common/Audio/SynthDef.sc

```diff
@@ -33,7 +33,7 @@ SynthDef {
 	}
 
 	*new { arg name, ugenGraphFunc, rates, prependArgs, variants, metadata;
-		^super.newCopyArgs(name.asSymbol).variants_(variants).metadata_(metadata).children_(Array.new(64))
+		^super.newCopyArgs(name.asSymbol).variants_(variants).metadata_(metadata ?? {()}).children_(Array.new(64))
```

Comment:
is this a (tiny) API change? Now having no metatdata won't test for `isNil` but rather `isEmpty`.

Reply:
It's a tiny API change, but shouldn't break any code (metadata can already be nil or non-nil, existing code should work in either case). This is a somewhat important change because without it, metadata has to be nil checked and lazily created for ALL internal points of access. This would be reasonable only if there was a significant cost to having a default-initialized metadata dict (there's not), or if there was somehow a special meaning to having a nil metadata (there's not, as far as I could find). Existing accesses to metadata in this class are fairly arcane because of the assumption that it can be nil - those can be significantly cleaned up (I didn't do that as part of this CL because it seems too unrelated). 
btw, happy to keep it at nil if it feels better, this just feels like a step towards better factoring.

Reply:
I can't check right now, but please test writing a SynthDef to disk to make sure that empty metadata aren't written to disk.

Currently metadata are written to disk in a second file (the scsyndef binary format is unchanged). If metadata are nil, no extra file is written (and any previously written metadata file will be removed).

A quick glance at the code convinces me that requiring a metadata object in all cases will produce 2 files for every SynthDef.

The relevant bits are in SynthDesc.sc.

Reply:
Ah, good eye, I didn't know that was happening.

Reply:
Tested and submitted a fix.